packages/react-native-reanimated/src/layoutReanimation/defaultTransitions/LinearStyleTransition.ts

```diff
@@ -0,0 +1,111 @@
+'use strict';
+import type { BaseAnimationBuilder } from '../animationBuilder';
+import { ComplexAnimationBuilder } from '../animationBuilder';
+import type {
+  ILayoutAnimationBuilder,
+  LayoutAnimationFunction,
+} from '../animationBuilder/commonTypes';
+
+/**
+ * Copy of LinearTransition supporting style transitions, Right now on
+ * experimental stage
+ */
+export class LinearStyleTransition
+  extends ComplexAnimationBuilder
+  implements ILayoutAnimationBuilder
+{
+  static presetName = 'LinearStyleTransition';
+
+  static createInstance<T extends typeof BaseAnimationBuilder>(
+    this: T
+  ): InstanceType<T> {
+    return new LinearStyleTransition() as InstanceType<T>;
+  }
+
+  build = (): LayoutAnimationFunction => {
+    const delayFunction = this.getDelayFunction();
+    const [animation, config] = this.getAnimationAndConfig();
+    const callback = this.callbackV;
+    const delay = this.getDelay();
+
+    return (values) => {
+      'worklet';
+      return {
```

Comment:
It seems better to generate an animation object based on properties provided in `values` instead of hardcoding them as one big object